ServerCore/Pages/Hints/Edit.cshtml.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Hints
+{
+    public class EditModel : EventSpecificPageModel
+    {
+        private readonly ServerCore.DataModel.PuzzleServerContext _context;
+
+        public EditModel(ServerCore.DataModel.PuzzleServerContext context)
+        {
+            _context = context;
+        }
+
+        [BindProperty]
+        public Hint Hint { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            Hint = await _context.Hints.FirstOrDefaultAsync(m => m.Id == id);
+
+            if (Hint == null)
+            {
+                return NotFound();
+            }
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            ModelState.Remove("Hint.Puzzle");
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            _context.Attach(Hint).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!HintExists(Hint.Id))
```

Comment:
Is all this handling actually needed? How bad would it be if you just removed the try/catch entirely?

Reply:
This is scaffolded boilerplate and makes error handling marginally nicer